lstchain/calib/camera/calibration_calculator.py

```diff
@@ -0,0 +1,258 @@
+        """
+        Calculate calibration coefficients from flatfield and pedestal statistics
+        associated to the present event
+
+        Parameters
+        ----------
+        event: EventAndMonDataContainer
+
+        """
+
+        ped_data = event.mon.tel[self.tel_id].pedestal
+        ff_data = event.mon.tel[self.tel_id].flatfield
+        status_data = event.mon.tel[self.tel_id].pixel_status
+        calib_data = event.mon.tel[self.tel_id].calibration
+
+        # mask from pedestal and flat-field data
+        monitoring_unusable_pixels = np.logical_or(status_data.pedestal_failing_pixels,
+                                                   status_data.flatfield_failing_pixels)
+        # calibration unusable pixels are an OR of all masks
+        calib_data.unusable_pixels = np.logical_or(monitoring_unusable_pixels,
+                                                   status_data.hardware_failing_pixels)
+        # Extract calibration coefficients with F-factor method
+        # Assume fix F2 factor, F2=1+Var(gain)/Mean(Gain)**2 must be known from elsewhere
+        F2 = 1.2
+
+        # calculate photon-electrons
+        n_pe = F2 * (ff_data.charge_median - ped_data.charge_median) ** 2 / (
+                ff_data.charge_std ** 2 - ped_data.charge_std ** 2)
+
+        # fill WaveformCalibrationContainer (this is an example)
+        calib_data.time = ff_data.sample_time
+        calib_data.time_range = ff_data.sample_time_range
+        calib_data.n_pe = n_pe
+
+        # find signal median of good pixels
+        masked_npe = np.ma.array(n_pe, mask=calib_data.unusable_pixels)
+        npe_signal_median = np.ma.median(masked_npe, axis=1)
+
+        # Flat field factor
+        ff = np.ma.getdata(npe_signal_median[:, np.newaxis] / n_pe)
+
+        calib_data.dc_to_pe = n_pe / (ff_data.charge_median - ped_data.charge_median) * ff
+
+        # put the time around zero
+        camera_time_median = np.median(ff_data.time_median, axis=1)
+        calib_data.time_correction = -ff_data.relative_time_median - camera_time_median[:, np.newaxis]
+
+        ped_extractor_name = self.config.get("PedestalCalculator").get("charge_product")
+        ped_width = self.config.get(ped_extractor_name).get("window_width")
+        calib_data.pedestal_per_sample = ped_data.charge_median / ped_width
+
+        # put to zero unusable pixels
+        calib_data.dc_to_pe[calib_data.unusable_pixels] = 0
+        calib_data.pedestal_per_sample[calib_data.unusable_pixels] = 0
+
+        # eliminate inf values id any (still necessary?)
+        calib_data.dc_to_pe[np.isinf(calib_data.dc_to_pe)] = 0
+
+
+
+    def process_interleaved(self, event):
+        """
+        Process interleaved calibration events (pedestals and FF)
+        Parameters
+        ----------
+        """
+        new_ped = False
+        new_ff = False
+
+        # if pedestal event
+        if event.r1.tel[self.tel_id].trigger_type == 32:
```

Comment:
It would probably be usefull to define a  dictionary with trigger types centrally somewhere instead of hard coding values here.

Reply:
yes, you are right, actually I promised @moralejo to do it. Perhaps the best place is in ctapipe_io_lst? (but that will ask to create a new release)

Reply:
Yes, this should be done then.

Reply:
I preferred to add a class in the lstcontainers.py module. I hope the trigger bit description is still correct